client.go

```diff
@@ -224,6 +225,9 @@ func decodeString(body io.Reader, output *string) error {
 // fullURL returns full URL for request.
 // args[0] is model name, if API type is Azure, model name is required to get deployment name.
 func (c *Client) fullURL(suffix string, args ...any) string {
+	if val := os.Getenv("OPENAI_BASEURL"); val != "" {
+		c.config.BaseURL = val
```

Comment:
The `fullURL` method is getting called during the usage of the client (days, weeks), so this change allows you to quietly change the base url using an environment variable from anywhere if you could just set an environment variable.

Reply:
I didn't realize that, my bad. I wanted to provide a non disruptive and quick way of supporting another API endpoint.  
Is alternative API endpoint support planned somehow or is this project only focusing at OpenAI?

Reply:
You probably want something like that:

```go
config := openai.DefaultConfig("token")
config.BaseURL = "your url" // probably "https://api.mistral.ai" 
client := openai.NewClientWithConfig(config)

response, err := client.CreateChatCompletion(
	context.Background(),
	openai.ChatCompletionRequest{
		Model: "mistral-small",
		Messages: []openai.ChatCompletionMessage{
			{
				Role:    openai.ChatMessageRoleUser,
				Content: "Hello!",
			},
		},
	},
)


```

Reply:
Ok I totally missed the `NewClientWithConfig()` function, thanks for clarifying.  
As per the `xz` vibes, I use `go-openai` in this little project of mine: https://gitlab.com/iMil/cbg  
Again, sorry for the misunderstanding and keep up the good work.